package-build.el

```diff
@@ -775,13 +775,15 @@ Optionally PRETTY-PRINT the data."
              extras))
         (error "No define-package found in %s" file-path)))))
 
-(defun pb/merge-package-info (pkg-info name version)
+(defun pb/merge-package-info (pkg-info name version dependencies)
   "Return a version of PKG-INFO updated with NAME, VERSION and info from CONFIG.
 If PKG-INFO is nil, an empty one is created."
   (let* ((merged (or (copy-sequence pkg-info)
                      (vector name nil "No description available." version))))
     (aset merged 0 name)
     (aset merged 3 version)
+    (when (not (aref merged 1))
+      (aset merged 1 dependencies))
```

Comment:
I'm not sure that it makes sense to *only* replace the dependencies if they don't exist. In fact if this is the case I don't think we should merge this.

For the behavior to make sense we should either make it so the dependencies are *union'd* or the dependencies are *replaced*.

As a first step I'm happy if you just replace the deps if they are specified.

```
(when dependencies
    (aset merged 1 dependencies))
```

and in the future we can discuss a merge.